sources/packages/backend/apps/api/src/route-controllers/note/note.institutions.controller.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  Controller,
+  Get,
+  Param,
+  Query,
+  ParseIntPipe,
+  NotFoundException,
+} from "@nestjs/common";
+import { StudentService } from "../../services";
+import BaseController from "../BaseController";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { NoteType } from "@sims/sims-db";
+import { NoteAPIOutDTO, transformToNoteDTO } from "./models/note.dto";
+import {
+  AllowAuthorizedParty,
+  IsBCPublicInstitution,
+} from "../../auth/decorators";
+import { ApiNotFoundResponse, ApiTags } from "@nestjs/swagger";
+import { ClientTypeBaseRoute } from "../../types";
+import { ParseEnumQueryPipe } from "../utils/custom-validation-pipe";
+
+/**
+ * Institution Controller for Notes.
+ */
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@IsBCPublicInstitution()
+@Controller("note")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-note`)
+export class NoteInstitutionsController extends BaseController {
+  constructor(private readonly studentService: StudentService) {
+    super();
+  }
+
+  /**
+   * Get notes for a student.
+   * @param studentId student id.
+   * @param noteType note type enum which is passed to filter the notes.
+   * @returns student notes.
+   */
+  @ApiNotFoundResponse({ description: "Student not found." })
+  @Get("student/:studentId")
+  async getStudentNotes(
+    @Param("studentId", ParseIntPipe) studentId: number,
+    @Query("noteType", new ParseEnumQueryPipe(NoteType)) noteType?: NoteType,
+  ): Promise<NoteAPIOutDTO[]> {
```

Comment:
looks like it is similar to ministry, probably we can create a controller and do the adjustments there. 
![image](https://github.com/bcgov/SIMS/assets/77353155/c3c0d12d-9902-45d2-aee3-7c1ec63008e8)

Reply:
I don't see benefits creating a controller service for this one. Thoughts? @andrewsignori-aot @dheepak-aot @guru-aot @sh16011993

Reply:
IMO I agree, Controller service are more like common helper methods created for multiple client types, here the actual logic is in getStudentNotes and it is defined differently in both the client types. I also feel, it can stay as it is.

Reply:
I do not see a problem to have the code centralized and take the opportunity to move the `transformToNoteDTO` to the controller service also.
Since this is about notes and it is not a complex code I wouldn't consider it as a blocker but I would still recommend having it done to be more aligned with the other controllers.

Reply:
With following piece of code applicable only for minstry controller, I would say controller service wouldn't be much useful, it will just replace the `transformToNoteDTO()`

```ts
 const student = await this.studentService.getStudentById(studentId);
    if (!student) {
      throw new NotFoundException("Student not found.");
    }
```